zeppelin-plugins/launcher/k8s-standard/src/main/java/org/apache/zeppelin/interpreter/launcher/K8sStandardInterpreterLauncher.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.interpreter.launcher;
+
+import java.io.File;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
+import org.apache.zeppelin.interpreter.recovery.RecoveryStorage;
+import org.apache.zeppelin.interpreter.remote.RemoteInterpreterUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Interpreter Launcher which use shell script to launch the interpreter process.
+ */
+public class K8sStandardInterpreterLauncher extends InterpreterLauncher {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(K8sStandardInterpreterLauncher.class);
+  private final Kubectl kubectl;
+  private InterpreterLaunchContext context;
+
+
+  public K8sStandardInterpreterLauncher(ZeppelinConfiguration zConf, RecoveryStorage recoveryStorage) throws IOException {
+    super(zConf, recoveryStorage);
+    kubectl = new Kubectl(zConf.getK8sKubectlCmd());
+    kubectl.setNamespace(getNamespace());
+  }
+
+  @VisibleForTesting
+  K8sStandardInterpreterLauncher(ZeppelinConfiguration zConf, RecoveryStorage recoveryStorage, Kubectl kubectl) {
+    super(zConf, recoveryStorage);
+    this.kubectl = kubectl;
+  }
+
+
+  /**
+   * Check if i'm running inside of kubernetes or not.
+   * @return
+   */
+  boolean isRunningOnKubernetes() {
```

Comment:
Is this really needed? It appears redundant since the same check is happening in 
[InterpreterSetting.java](https://github.com/apache/zeppelin/blob/22e090f6197d907764b4237d7934eb4d8a19c18d/zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java#L639)

Reply:
Yes, it is. I added a comment in the source code why we need `isRunningOnKubernetes()` inside of `K8sStandardInterpreterLauncher`, although `K8sStandardInterpreterLauncher` supposed to be used only in Kubernetes cluster which `InterpreterSetting` supposed to detect beforehand.

Reply:
Thanks for the explanation. It is helpful.